src/text-input/index.ts

```diff
@@ -47,10 +48,11 @@ export interface TextInputProperties extends ThemedProperties, FocusProperties,
 	helperText?: string;
 	value?: string;
 	valid?: { valid?: boolean; message?: string; } | boolean;
+	customValidator?: (value: string) => { valid?: boolean; message?: string; } | void;
 	pattern?: string | RegExp;
 	autocomplete?: boolean | string;
 	onClick?(value: string): void;
-	onValidate?: (valid: boolean, message?: string) => void;
+	onValidate?: (valid?: boolean, message?: string) => void;
```

Comment:
is it correct that this is optional? I thought that it would always be passed but it may be `undefined`, in which case perhaps it should be `onValidate?: (valid: boolean | undefined, message?: string) => void;` ?